File: quartz-energy/src/backend/QuartzEnergy.FracSchedule.Services/FracSchedule/Models/Schedules/Overviews/ScheduleOverviews.cs
namespace QuartzEnergy.FracSchedule.Services.FracSchedule.Models.Schedules.Overviews
{
    using System.Collections.Generic;

    using QuartzEnergy.Common.Services.Models.Overviews;

    public sealed class ScheduleOverviews : OverviewsModel<ScheduleOverviewsRequest, ScheduleOverview>
    {
        public ScheduleOverviews(
            ScheduleOverviewsRequest request,
            int recordsCount,
            IEnumerable<ScheduleOverview> overviews,
            ScheduleSummary summary)
            : base(request, recordsCount, overviews)
        {
            this.Summary = summary;
        }

        public ScheduleSummary Summary { get; }
    }
}

File: quartz-energy/src/frontend/src/app/frac-schedule/models/schedules/overviews/schedule-overviews.ts
import { ScheduleOverviewsRequest } from './schedule-overviews-request';
import { Overviews } from '../../../../common/models/overviews/overviews';
import { ScheduleOverview } from './schedule-overview';
import { ScheduleSummary } from './schedule-summary';

export class ScheduleOverviews extends Overviews<ScheduleOverviewsRequest, ScheduleOverview> {
    constructor(
      public request: ScheduleOverviewsRequest,
      public recordsCount: number,
      public overviews: ScheduleOverview[],
      public summary: ScheduleSummary) {
      super(request, recordsCount, overviews);
    }
}
